tests/test_routes.py

```diff
@@ -115,6 +147,66 @@ def test_update_fund(flask_test_client, seed_dynamic_data):
         elif key != "submit":
             assert updated_fund.__getattribute__(key) == value
 
+@pytest.mark.usefixtures("set_auth_cookie", "patch_validate_token_rs256_internal_user")
+def test_create_round_with_existing_short_name(flask_test_client, seed_dynamic_data):
+    """
+    Tests that a round can be successfully created using the /round route
+    Verifies that the created round has the correct attributes
+    """
+    test_fund = seed_dynamic_data["funds"][0]
+    new_round_data = {
+        "fund_id": test_fund.fund_id,
+        "title_en": "New Round",
+        "short_name": "NR123",
+        "opens-day": "01",
+        "opens-month": "10",
+        "opens-year": "2024",
+        "opens-hour": "09",
+        "opens-minute": "00",
+        "deadline-day": "01",
+        "deadline-month": "12",
+        "deadline-year": "2024",
+        "deadline-hour": "17",
+        "deadline-minute": "00",
+        "assessment_start-day": "02",
+        "assessment_start-month": "12",
+        "assessment_start-year": "2024",
+        "assessment_start-hour": "09",
+        "assessment_start-minute": "00",
+        "reminder_date-day": "15",
+        "reminder_date-month": "11",
+        "reminder_date-year": "2024",
+        "reminder_date-hour": "09",
+        "reminder_date-minute": "00",
+        "assessment_deadline-day": "15",
+        "assessment_deadline-month": "12",
+        "assessment_deadline-year": "2024",
+        "assessment_deadline-hour": "17",
+        "assessment_deadline-minute": "00",
+        "prospectus_link": "http://example.com/prospectus",
+        "privacy_notice_link": "http://example.com/privacy",
+        "contact_email": "contact@example.com",
+        "submit": "Submit",
+        "contact_phone": "1234567890",
+        "contact_textphone": "0987654321",
+        "support_times": "9am - 5pm",
+        "support_days": "Monday to Friday",
+        "feedback_link": "http://example.com/feedback",
+        "project_name_field_id": 1,
+        "guidance_url": "http://example.com/guidance",
+    }
+
+    response = submit_form(flask_test_client, "/round", new_round_data)
+    assert response.status_code == 200
+    new_round_data = {**new_round_data, 'short_name': 'NR1234'}
+    response = submit_form(flask_test_client, "/round", new_round_data)
+    assert response.status_code == 200
```

Comment:
Could we have assertions to check the error message doesn't appear in the HTML for the first two submissions? As we expect a 200 response either way. We could have:

```
error_html = '<a href="#short_name">Short name: Given short name already exists in the fund funding to improve testing.</a>'

# Test works fine with first round
response = submit_form(flask_test_client, "/round", new_round_data)
assert response.status_code == 200
assert error_html not in response.data.decode('utf-8'), "Error HTML found in response"

# Test works fine with second round but with different short name
new_round_data = {**new_round_data, 'short_name': 'NR1234'}
response = submit_form(flask_test_client, "/round", new_round_data)
assert response.status_code == 200
assert error_html not in response.data.decode('utf-8'), "Error HTML found in response"

# Test doesn't work with third round with same short name as firsrt
new_round_data = {**new_round_data, 'short_name': 'NR123'}
response = submit_form(flask_test_client, "/round", new_round_data)
assert response.status_code == 200
assert error_html in html = response.data.decode('utf-8'), "Error HTML not found in response"
```